waiter/integration/waiter/instance_tracker_integration_test.clj

```diff
@@ -0,0 +1,80 @@
+(ns waiter.instance-tracker-integration-test
+  (:require [clojure.set :as set]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [clojure.walk :as walk]
+            [waiter.status-codes :refer :all]
+            [waiter.util.client-tools :refer :all]))
+
+(defn- get-instance-tracker-state
+  [waiter-url & {:keys [cookies keywordize-keys query-params]
+                 :or {keywordize-keys true}}]
+  (let [response (make-request waiter-url "/state/instance-tracker"
+                               :cookies cookies
+                               :query-params query-params
+                               :method :get)]
+    (assert-response-status response http-200-ok)
+    (cond-> (some-> response :body try-parse-json)
+            keywordize-keys walk/keywordize-keys)))
+
+(deftest ^:parallel ^:integration-fast test-instance-tracker-daemon-state
+  (testing-using-waiter-url
+    (let [{:keys [cookies]} (make-request waiter-url "/waiter-auth")]
+
+      (testing "no query parameters provides the default state fields"
+        (let [{{:keys [supported-include-params] :as state} :state} (get-instance-tracker-state waiter-url :cookies cookies)
+              default-state-fields #{:last-update-time :supported-include-params}]
+          (is (= (set (keys state))
+                 default-state-fields))
+          (is (= (set supported-include-params)
+                 #{"id->failed-instance" "instance-failure-handler"}))))
+
+      (testing "InstanceEventHandler provides default state fields"
+        (let [query-params "include=instance-failure-handler"
+              body (get-instance-tracker-state waiter-url
+                                               :cookies cookies
+                                               :query-params query-params)
+              default-inst-event-handler-state-fields #{:last-error-time :supported-include-params :type}]
+          (is (set/superset? (set (keys (get-in body [:state :instance-failure-handler] #{"!@#!@#14132"})))
+                             default-inst-event-handler-state-fields)))))))
+
+(deftest ^:parallel ^:integration-fast ^:resource-heavy test-instance-tracker-failing-instance
+  (testing-using-waiter-url
+    (let [{:keys [cookies]} (make-request waiter-url "/waiter-auth")]
+
+      ; TODO: When instance tracking becomes more reliable in multi router scenarios this test should not be
+      ; on the same router
+      (testing "new failing instances appear in instance-tracker state on same router"
+        (let [router-url (rand-router-url waiter-url)
```

Comment:
we should check on all routers.

kitchen has a header that causes a healthy instance to die, we could be using that too.